src/App.vue

```diff
@@ -34,7 +34,7 @@
     </v-main>
 
     <!-- auth guard -->
-    <AuthenticationGuard v-if="!isAuthenticated" />
+    <AuthenticationGuard v-if="!isAuthenticated || (isAuthenticated && !current_user.emailVerified)" />
```

Comment:
I would change the approach here a little. Instead adding more complex condition here and other templates, why don't we embed this condition into the getter of isAuthenticated. Then you don't have to make individual changes in the each template but rather only in the pinia store.